.build/build.ps1

```diff
@@ -0,0 +1,191 @@
+    "AssemblyFileVersion = $script:AssemblyFileVersion"
+    "NugetPackageVersion = $script:NugetPackageVersion"
+    "ReleaseNotes = $script:ReleaseNotes"
+}
+
+# Synopsis: Restore the nuget packages of the Visual Studio solution
+task RestoreNugetPackages {
+    exec {
+        & $NugetExe restore "$Solution" -Verbosity detailed
+    }
+}
+
+# Synopsis: Update the nuget packages of the Visual Studio solution
+task UpdateNugetPackages RestoreNugetPackages, {
+    exec {
+        & $NugetExe update "$Solution" -Verbosity detailed
+    }
+}
+
+# Synopsis: Update the version info in all AssemblyInfo.cs
+task UpdateVersionInfo GenerateVersionInformation, {
+    "Updating assembly information"
+
+    # Ignore anything under the Testing/ folder
+    @(Get-ChildItem "$RootDir" AssemblyInfo.cs -Recurse) | where { $_.FullName -notlike "$RootDir\Testing\*" } | ForEach {
+        Update-AssemblyVersion $_.FullName `
+            -Version $script:AssemblyVersion `
+            -FileVersion $script:AssemblyFileVersion `
+            -InformationalVersion $script:NuGetPackageVersion
+    }
+}
+
+# Synopsis: Update the nuspec dependencies versions based on the versions of the nuget packages that are being used
+task UpdateNuspecVersionInfo {
+    # Find all the packages.config
+    $projectFiles = Get-ChildItem "$RootDir" -Recurse -Filter "*.csproj" `
+                      | ?{ $_.fullname -notmatch "\\(.build)|(packages)\\" } `
+                      | Resolve-Path
+
+    # Update dependency verions in each of our .nuspec file based on what is in our packages.config
+    Resolve-Path "$RootDir\Nuspec\*.nuspec" | Update-NuspecDependenciesVersions -ProjectFilePaths $projectFiles -SpecificVersions -verbose
+}
+
+# Synopsis: A task that makes sure our initialization tasks have been run before we can do anything useful
+task Init CreateFolders, RestoreNugetPackages, GenerateVersionInformation
+
+# Synopsis: Compile the Visual Studio solution
+task Compile Init, UpdateVersionInfo, {
+    use 15.0 MSBuild
+    try {
+        exec {
+            & MSBuild `
+                "$Solution" `
+                /maxcpucount `
+                /nodereuse:false `
+                /target:Build `
+                /p:Configuration=$Configuration `
+                /flp1:verbosity=normal`;LogFile=$LogsDir\_msbuild.log.normal.txt `
+                /flp2:WarningsOnly`;LogFile=$LogsDir\_msbuild.log.warnings.txt `
+                /flp3:PerformanceSummary`;NoSummary`;verbosity=quiet`;LogFile=$LogsDir\_msbuild.log.performanceSummary.txt `
+                /flp4:verbosity=detailed`;LogFile=$LogsDir\_msbuild.log.detailed.txt `
+                /flp5:verbosity=diag`;LogFile=$LogsDir\_msbuild.log.diag.txt `
+        }
+    } finally {
+        TeamCity-PublishArtifact "$LogsDir\_msbuild.log.* => logs/msbuild.$Configuration.logs.zip"
+    }
+}
+
+# Synopsis: Execute our unit tests
+task UnitTests {
+    "$RootDir\bin\net461\*.Test.dll" | Resolve-Path | ForEach-Object {
```

Comment:
This is weird. Is everything building to a `bin` folder in the root of the working directory? We've been horribly stung by this sort of thing on the past. The preference these days is to adopt the default conventions as much as possible. In this case, it means building all projects into their own `bin\$Configuration`\net461` folder and running the tests from there.

Reply:
I just left it as it was for now. I'll fix that.